From: Rick Jones <rick.jones2@hp.com>
Date: Mon, 21 Nov 2011 10:54:05 +0000
Subject: [PATCH 13/24] Sweep away N/A fw_version dustbunnies from the
 .get_drvinfo routine of a number of drivers
Git-commit: 84b405011166e663fe9ef56c29b1d76f59b35568
Patch-mainline: v3.3-rc1
References: bnc#784800 FATE#314296

Per discussion with Ben Hutchings and David Miller, go through and
remove assignments of "N/A" to fw_version in various drivers'
.get_drvinfo routines.  While there clean-up some use of bare
constants and such.

Signed-off-by: Rick Jones <rick.jones2@hp.com>
Signed-off-by: David S. Miller <davem@davemloft.net>
[limited to chelsio drivers]
Acked-by: Benjamin Poirier <bpoirier@suse.de>
---
 drivers/net/ethernet/chelsio/cxgb/cxgb2.c       |    1 -
 drivers/net/ethernet/chelsio/cxgb3/cxgb3_main.c |    5 +----
 drivers/net/ethernet/chelsio/cxgb4/cxgb4_main.c |    4 +---
 3 files changed, 2 insertions(+), 8 deletions(-)

--- a/drivers/net/ethernet/chelsio/cxgb/cxgb2.c
+++ b/drivers/net/ethernet/chelsio/cxgb/cxgb2.c
@@ -434,7 +434,6 @@ static void get_drvinfo(struct net_devic
 
 	strlcpy(info->driver, DRV_NAME, sizeof(info->driver));
 	strlcpy(info->version, DRV_VERSION, sizeof(info->version));
-	strlcpy(info->fw_version, "N/A", sizeof(info->fw_version));
 	strlcpy(info->bus_info, pci_name(adapter->pdev),
 		sizeof(info->bus_info));
 }
--- a/drivers/net/ethernet/chelsio/cxgb3/cxgb3_main.c
+++ b/drivers/net/ethernet/chelsio/cxgb3/cxgb3_main.c
@@ -1580,9 +1580,7 @@ static void get_drvinfo(struct net_devic
 	strlcpy(info->version, DRV_VERSION, sizeof(info->version));
 	strlcpy(info->bus_info, pci_name(adapter->pdev),
 		sizeof(info->bus_info));
-	if (!fw_vers)
-		strlcpy(info->fw_version, "N/A", sizeof(info->fw_version));
-	else {
+	if (fw_vers)
 		snprintf(info->fw_version, sizeof(info->fw_version),
 			 "%s %u.%u.%u TP %u.%u.%u",
 			 G_FW_VERSION_TYPE(fw_vers) ? "T" : "N",
@@ -1592,7 +1590,6 @@ static void get_drvinfo(struct net_devic
 			 G_TP_VERSION_MAJOR(tp_vers),
 			 G_TP_VERSION_MINOR(tp_vers),
 			 G_TP_VERSION_MICRO(tp_vers));
-	}
 }
 
 static void get_strings(struct net_device *dev, u32 stringset, u8 * data)
--- a/drivers/net/ethernet/chelsio/cxgb4/cxgb4_main.c
+++ b/drivers/net/ethernet/chelsio/cxgb4/cxgb4_main.c
@@ -1006,9 +1006,7 @@ static void get_drvinfo(struct net_devic
 	strlcpy(info->bus_info, pci_name(adapter->pdev),
 		sizeof(info->bus_info));
 
-	if (!adapter->params.fw_vers)
-		strlcpy(info->fw_version, "N/A", sizeof(info->fw_version));
-	else
+	if (adapter->params.fw_vers)
 		snprintf(info->fw_version, sizeof(info->fw_version),
 			"%u.%u.%u.%u, TP %u.%u.%u.%u",
 			FW_HDR_FW_VER_MAJOR_GET(adapter->params.fw_vers),
